lib/linter/apply-disable-directives.js

```diff
@@ -443,17 +450,33 @@ module.exports = ({ directives, disableFixes, problems, reportUnusedDisableDirec
         }
     }).sort(compareLocations);
 
+    // This determines a list of rules that are not being run by the given ruleFilter, if present.
+    const rulesToIgnore = configuredRules && ruleFilter
+        ? new Set(Object.keys(configuredRules).filter(ruleId => {
+            const severity = ConfigOps.getRuleSeverity(configuredRules[ruleId]);
+
+            return !ruleFilter({ severity, ruleId });
```

Comment:
```suggestion
            const severity = ConfigOps.getRuleSeverity(configuredRules[ruleId]);
            
            if (severity === 0) {
                return false;
            }

            return !ruleFilter({ severity, ruleId });
```

Rules that are turned off should not be ignored when checking unused disable directives (and generally shouldn't be passed to `ruleFilter()`).

```js
/* eslint no-undefined: 0 */

/* eslint-disable no-undefined */   // <- should be reported as unused 

```